### DISCUSSION TITLE
Replace EEM or CPCIS connector/s with SFP

### DISCUSSION THREAD

**Discussion by vnegnev** [sinara-hw/EEM_FMC_Carrier#59] [Specification]
Hi, we're looking into using the EEM-FMC carrier in an application where two SFP cages would be useful - one to support Ethernet and the other for optical-fibre DRTIO.

Based on [this earlier discussion](https://github.com/sinara-hw/EEM_FMC_Carrier/discussions/12#discussioncomment-1693404) between @gkasprow and @marmeladapk it sounds like there was going to be an add-in board that provides an SFP cage, but I haven't found any other details about it. The alternative seemed to be using a SATA-to-SFP adapter board but it doesn't look like the SATA ports made it into the design - perhaps I'm missing something. In any case a more closely integrated adapter PCB not requiring external power seems nicer mechanically and from a signal integrity point of view.

What do people think about making a variant of the board supporting 1 or 2 SFPs? Would this be a lot of design work, and would other people be interested? What is the status of the SFP add-in board?

**Comment by gkasprow** [comment]
I think we were discussing this addon board for KASLI-DIOT; there is a connector for such board.
Here we could porobably fit 1 SFP cage (or stacked SFP cage); but we have to wait for another revision.

**Comment by gkasprow** [comment]
@vnegnev what MGT speed are you aiming for?
maybe it's enough to just solder one side of SFP cable directly to DIOT connector pads
![image](https://github.com/sinara-hw/EEM_FMC_Carrier/assets/4325054/3a83196b-9880-41a2-b47c-91b6497b4464)

**Reply by vnegnev** [reply]
> what MGT speed are you aiming for?

Short answer is 'not too fast', long answer is 'not too fast now but we want to avoid it being a bottleneck if we want to pump more data through it in the future'.

> maybe it's enough to just solder one side of SFP cable directly to DIOT connector pads

For testing/development that's probably fine, however a big motivation for us is to avoid mods to the PCBs if possible - the aim is to have a board that a technician without much electronics knowledge can install/test/swap out.

**Comment by gkasprow** [comment]
Sure, we will either add SATA or SFP, whatever fits.  Some users need additional high-speed port for HPC projects
